HC/WrapperGen/WrapperGen.cpp

```diff
@@ -489,22 +502,22 @@ struct StringFinder
             func->printArgsAsInitializers(out);
           }
           out << "{\n";
-          out << "_lp.gridDim.x = __lp.gridDim.x;\n";
-          out << "_lp.gridDim.y = __lp.gridDim.y;\n";
-          out << "_lp.gridDim.z = __lp.gridDim.z;\n";
-          out << "_lp.groupDim.x = __lp.groupDim.x;\n";
-          out << "_lp.groupDim.y = __lp.groupDim.y;\n";
-          out << "_lp.groupDim.z = __lp.groupDim.z;\n";
-          out << "_lp.groupMemBytes = __lp.groupMemBytes;\n";
+          out << "_lp.grid_dim.x = x_lp.grid_dim.x;\n";
+          out << "_lp.grid_dim.y = x_lp.grid_dim.y;\n";
+          out << "_lp.grid_dim.z = x_lp.grid_dim.z;\n";
+          out << "_lp.group_dim.x = x_lp.group_dim.x;\n";
+          out << "_lp.group_dim.y = x_lp.group_dim.y;\n";
+          out << "_lp.group_dim.z = x_lp.group_dim.z;\n";
+          out << "_lp.dynamic_group_mem_bytes = x_lp.dynamic_group_mem_bytes;\n";
+#if CHECK_LP_ARGS
+          out << "assert(x_lp.barrier_bit   ==  barrier_bit_queue_default);\n";  // remove when barrier-bit supported
+          out << "assert(x_lp.launch_fence  == -1);\n";  // remove when launch_fence supported
           out << "}\n";
+#endif
+
 
+          // This code is in the compute kernel that is executed on the accelerator :
           out << "void operator()(tiled_index<3>& i) __attribute((hc))\n{\n";
-          out << "_lp.groupId.x = i.tile[2];\n";
-          out << "_lp.groupId.y = i.tile[1];\n";
-          out << "_lp.groupId.z = i.tile[0];\n";
-          out << "_lp.threadId.x = i.local[2];\n";
-          out << "_lp.threadId.y = i.local[1];\n";
-          out << "_lp.threadId.z = i.local[0];\n";
```

Comment:
Yes exactly.  Program can use built in "coordinate functions" instead.


On Jun 22, 2016, at 8:03 PM, Wen-Heng (Jack) Chung <notifications@github.com<mailto:notifications@github.com>> wrote:


In HC/WrapperGen/WrapperGen.cpp<https://github.com/RadeonOpenCompute/hcc/pull/80#discussion_r68161840>:

>            out << "void operator()(tiled_index<3>& i) __attribute((hc))\n{\n";
> -          out << "_lp.groupId.x = i.tile[2];\n";
> -          out << "_lp.groupId.y = i.tile[1];\n";
> -          out << "_lp.groupId.z = i.tile[0];\n";
> -          out << "_lp.threadId.x = i.local[2];\n";
> -          out << "_lp.threadId.y = i.local[1];\n";
> -          out << "_lp.threadId.z = i.local[0];\n";


these lines are no longer populated in grid_paunch_parm because kernel could access these values from other utility functions, right?

-
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/RadeonOpenCompute/hcc/pull/80/files/758b7f57d57a3a927dd454b7d8011318ecb129f0#r68161840>, or mute the thread<https://github.com/notifications/unsubscribe/ACYSAujUxBYVsO6m6t9SDxiN01RnwS29ks5qOdtqgaJpZM4I8LAJ>.